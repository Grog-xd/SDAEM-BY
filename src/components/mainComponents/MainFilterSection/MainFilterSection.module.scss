@import "../../../style/mixins";
.filterSection{
  @supports (background: url("../../../assets/img/bg-404.webp")) {
    background: url("../../../assets/img/bg-404.webp") center center/cover no-repeat, linear-gradient(0deg, #7E6AF3 1.04%, #6540CD 100%);
  }
  background: url("../../../assets/img/bg-404.png") center center/cover no-repeat, linear-gradient(0deg, #7E6AF3 1.04%, #6540CD 100%);
  padding-bottom: 120px;
  border-radius: 0 0 40px 40px;
  padding-top: 83px;
  h1{
    @include text(30px, 700, 36px, #fff);
    text-shadow: 0 2px 5px rgba(0, 14, 142, 0.3);
    text-align: center;
    margin-bottom: 50px;
    b{
      color: #FFD54F;
    }
  }
}


.mainForm{
  max-width: 1032px;
  padding: 0 15px;
  margin: 0 auto;
}


.inputBLock{
  margin: 15px 0 20px 20px;
  padding-right: 17px;
  border-right: 1px solid rgba(102, 78, 249, 0.1);
  .inputBLockText{
    @include text(14px, 500, 17px, #BDBDBD);
    margin-bottom: 11px;
  }
}


.costBlock{
  display: flex;
  justify-content: space-between;
  align-items: center;
  color: #4E64F9;
  //input[type="number"]::-webkit-outer-spin-button,
  //input[type="number"]::-webkit-inner-spin-button {
  //  display: none;
  //}
  input{
    background: #F8F8F8;
    border-radius: 20px;
    border: none;
    padding: 9px 14px 11px 14px;
    width: 72px;
    outline: none;
    &:first-child{
      margin-right: 4px;
    }
    &:last-child{
      margin-left: 4px;
    }
    &::placeholder{
      @include text(14px, 500, 17px, #686868)
    }
  }
}

.openMoreFilter{
  display: flex;
  align-items: center;
  @include text(14px, 500, 17px, #1E2123);
  border: none;
  background: transparent;
  cursor: pointer;
  padding: 20px 0;
  svg{
    margin-left: 4px;
  }
}

.openMapButton{
  @include text(14px, 500, 17px, #1E2123);
  display: flex;
  align-items: center;
  border: none;
  padding-left: 17px;
  padding-right: 17px;
  svg{
    margin-left: 5px;
  }
}

.filterButton{
  display: flex;
  align-items: center;
  padding: 9px 16px;
  @include text(14px, 700, 17px, #242424);
  background: #FFD54F;
  border-radius: 22px;
  margin-right: 34px;
  cursor: pointer;
  outline: none;
  border: none;
  &:hover{
    background: #FEC100;
  }
  svg{
    margin-left: 12px;
  }
}
