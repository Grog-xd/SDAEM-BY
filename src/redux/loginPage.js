import {createSlice} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, setBookMarkActive, registerUser} = loginPage.actions

