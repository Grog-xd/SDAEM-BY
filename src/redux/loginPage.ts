import {createSlice, PayloadAction} from '@reduxjs/toolkit';

import avatar from '../assets/img/avatar.png'
import {profilesArr} from '../types/types';

const loginPage = createSlice({
    name: 'login',
    initialState:{
        // bookMarkActive:false,
        isAuth: false,
        profilesArr:[{name: 'dmitriy', avatar:'', phone:'', city:'', email: 'vladimir6234@tut.by', password: '12345', viber: '', whatsUpp:''},{name: 'test', avatar:avatar, phone:'+test', city:'test', email: 'test@tut.by', password: 'test1', viber: 'test', whatsUpp:'test'}],
        profile: {},
    },
    reducers:{
        // setBookMarkActive(state){
        //     state.bookMarkActive = !state.bookMarkActive
        // },
        enter(state, action:PayloadAction<profilesArr>){
            state.isAuth = true
            state.profile = action.payload
        },
        exit(state){
            state.isAuth = false
            localStorage.clear()
        },
        registerUser(state, action:PayloadAction<any>){
            state.profilesArr = [...state.profilesArr, action.payload]
        },
    },
})

export default loginPage.reducer
export const {exit, enter, registerUser} = loginPage.actions

