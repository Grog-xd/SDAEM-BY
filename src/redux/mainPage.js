import {createSlice} from '@reduxjs/toolkit';

const main = createSlice({
    name: 'main',
    initialState:{
        tabs:[{id:'flats', value:'Квартиры'}, {id:'cottages', value:'Котеджы и усадьбы'}, {id:'baths', value:'Бани и сауны'},  {id:'cars', value:'Авто на прокат'}],

        cityOption:[{id:1,value:'Минск'},{id:2,value:'Гомель'},{id:3,value:'Брест'},{id:4,value:'Витебск'},{id:5,value:'Гродно'}, {id:6,value:'Могилев'}],
        cityCurrentValue:'',

        roomsOption: [{id:1,value:1},{id:2,value:2},{id:3,value:3},{id:4,value:4},{id:5,value:5}],
        roomsCurrentValue:'',

        metroOption:[{id:1,value:'Грушевка'},{id:2,value:'Восход'},{id:3,value:'Пушкинская'},{id:4,value:'Партизанская'}],
        metroCurrentValue:'',

        districtOption:[{id:1,value:'Шабаны'},{id:2,value:'Запад'},{id:3,value:'Восток'},{id:4,value:'Малиновка'}],
        districtCurrentValue: '',

        productsLimit: 6,
        currentPage: 1,

        sortedOptions:[{id:1, value:'Сначала дешевые'}, {id:2, value: 'Сначала дорогие'}],
        sortedValue:'',


        bedSumOptions:[{id:1, value:'1'}, {id:2, value:'2'}, {id:3, value:'3'}, {id:4, value:'4'}, {id:5, value:'5'}],
        bedSumCurrentValue:'',

        checkboxsMoreOption:[{id:1, value:'Газовая плита', bol:false},  {id:2, value:'Духовка', bol:false}, {id:3, value:'Кофеварка', bol:false}, {id:4, value:'Микроволновая печь ', bol:false}, {id:5, value:'Посуда', bol:false}, {id:6, value:'Посудомоечная машина', bol:false}],

        minCost:'',
        maxCost:'',


        productsLen: 0,
        sortedProducts:[],
        products: [],
    },
    reducers: {
        fetchProducts(state, action){
            state.products = action.payload
        },
        setBedSum(state, action){
            state.bedSumCurrentValue = action.payload
        },

        checkboxHandler(state, action){
            state.checkboxsMoreOption.forEach(obj =>{
                if(obj.value.includes(action.payload)){
                    obj.bol = !obj.bol

                }
            })
        },

        likeHandler(state, action){
            state.products.forEach(obj=>{
                Object.keys(obj).forEach(key=>{
                    if(key === action.payload.type){
                        obj[key].forEach(e=>{
                            if (e.id === action.payload.id){
                                e.like = !e.like
                            }
                        })
                    }
                })
            })
        },
        setCity(state, action){
            state.cityCurrentValue = action.payload

        },
        setRooms(state, action){
            state.roomsCurrentValue = action.payload
        },
        setMinCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.minCost = action.payload
        },
        setMaxCost(state, action){
            if(action.payload<0 || action.payload > 1000){
                return
            }
            state.maxCost = action.payload
        },
        setDistrict(state, action){
            state.districtCurrentValue = action.payload
        },
        setMetro(state, action){
            state.metroCurrentValue = action.payload
        },
        setCurrentPage(state, action){
            state.currentPage = action.payload
            window.scrollTo(0, 0)
        },
        setSortedValue(state, action){
            state.sortedValue = action.payload
        },

        setLimit(state, action){
            state.productsLimit = action.payload
        },
        resetFilter(state){
            state.sortedValue = ''
            state.roomsCurrentValue = ''
            state.minCost = ''
            state.maxCost = ''
            state.districtCurrentValue = ''
            state.metroCurrentValue = ''
            state.bedSumCurrentValue = ''
            state.checkboxsMoreOption.forEach(e=>{
                e.bol = false
            })
        },

        mainFilter(state, action){
            // Условия для высчитывания стоимости
            while(state.minCost > state.maxCost ){
                state.maxCost = +state.maxCost + 100
            }
            if(state.minCost === ''){
                state.minCost=0
            }
            if(state.maxCost === ''){
                state.maxCost = 1000
            }


            //Основная фильтрация
            let sortedRes= []
            state.products.forEach(obj =>{
                if (action.payload){
                    Object.keys(obj).forEach(key =>{
                        if(action.payload === key){
                            obj[key].forEach(e=>{
                                if(!state.roomsCurrentValue){
                                    if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                        sortedRes.push(e)
                                    }
                                } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            })
                        }
                    })
                } else{
                    Object.keys(obj).forEach(key =>{
                        obj[key].forEach(e=>{
                            if(!state.roomsCurrentValue){
                                if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                    sortedRes.push(e)
                                }
                            } else if(+e.cost >= state.minCost && +e.cost <= state.maxCost && e.city.includes(state.cityCurrentValue) && e.rooms===state.roomsCurrentValue && e.district.includes(state.districtCurrentValue) && e.metro.includes(state.metroCurrentValue) && e.people.includes(state.bedSumCurrentValue)){
                                sortedRes.push(e)
                            }
                        })
                    })
                }
            })
            state.productsLen = sortedRes.length

            // Сортировка массива
            switch (state.sortedValue) {
            case 'Сначала дешевые':
                sortedRes.sort((a, b)=>{
                    return a.cost - b.cost
                })
                break
            case 'Сначала дорогие':
                sortedRes.sort((a, b)=>{
                    return b.cost - a.cost
                })
                break
            default:
                break
            }

            // Фильтрация по страницам
            let res = []
            let test = 0
            for (let i = 0; test < sortedRes.length; i++) {
                let obj = []
                for (let j = 0; j< state.productsLimit; j++) {
                    if(sortedRes[test]){
                        obj.push(sortedRes[test])
                        test++
                    }

                }
                res.push(obj)
            }
            state.sortedProducts = res
            state.currentPage = 1
        },

    },
})

export default main.reducer
export const {setCity, setRooms, setMinCost, setMaxCost, mainFilter, setDistrict, setMetro, resetFilter, setCurrentPage, setSortedValue, setLimit, likeHandler, setBedSum, checkboxHandler, fetchProducts} = main.actions
