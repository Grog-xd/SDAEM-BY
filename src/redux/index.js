import {combineReducers} from "redux";
import {configureStore} from "@reduxjs/toolkit";
import mainPage from "./mainPage";
import newsPage from "./newsPage";
import loginPage from "./loginPage";

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage
})

export const store = configureStore({
    reducer: rootReducer
})
