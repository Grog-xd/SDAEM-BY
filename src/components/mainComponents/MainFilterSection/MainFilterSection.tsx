import {FC, useState} from 'react';
import {Tab, TabList, TabPanel, Tabs} from 'react-tabs';
import {Link, useNavigate} from 'react-router-dom';
import {useDispatch, useSelector} from 'react-redux';

import {mainFilter, setCity, setMaxCost, setMinCost, setRooms} from '../../../redux/mainPage';
import MySelect from '../../UI/mySelect/mySelect';
import MoreOptions from '../../moreOptions/MoreOptions';
import SvgSettings from '../../svg/SvgSettings';
import SvgLocation from '../../svg/SvgLocation';
import SvgArrowRight from '../../svg/SvgArrowRight';

import {mainRedux} from '../../../types/types';

import classes from './MainFilterSection.module.scss';

const MainFilterSection:FC = () => {
    const dispatch = useDispatch()
    const navigate = useNavigate()
    const [tabIndex, setTabIndex] = useState<number>(0);
    const [moreOptions, setMoreOptions] = useState<boolean>(false)
    const {cityCurrentValue, roomsCurrentValue, roomsOption, cityOption, minCost, maxCost} = useSelector((state:mainRedux)=> state.main)

    function catalogFilterHandler(link:string){
        dispatch(mainFilter(link))
        navigate(`/catalog/${link}`)
    }

    return (
        <section data-testid={'main-filter-section'} className={classes.filterSection}>
            <h1>Sdaem.by - у нас живут <b>ваши объявления</b></h1>
            <form className={classes.mainForm}>
                <Tabs  defaultIndex={tabIndex} onSelect={( index) => setTabIndex(index)}>
                    <TabList>
                        <Tab data-testid={'tab-btn'}>Квартиры на сутки</Tab>
                        <Tab data-testid={'tab-btn'}>Коттеджи и усадьбы</Tab>
                        <Tab data-testid={'tab-btn'}>Бани и сауны</Tab>
                        <Tab data-testid={'tab-btn'}>Авто напрокат</Tab>
                    </TabList>
                    <TabPanel data-testid={'tab-panel'}>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Город</p>
                            <MySelect options={cityOption} handler={(e)=>dispatch(setCity(e))} value={cityCurrentValue ? cityCurrentValue : 'Выберите'}/>
                        </div>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Комнаты</p>
                            <MySelect options={roomsOption} handler={(e)=>dispatch(setRooms(e))} value={roomsCurrentValue ? roomsCurrentValue : 'Выберите'}/>
                        </div>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Цена за сутки (BYN)</p>
                            <div className={classes.costBlock}>
                                <input type='number' min={0} max={1000} value={minCost} placeholder={'От'} onChange={(e)=> dispatch(setMinCost(+e.target.value))}/>
                                -
                                <input type='number' min={0} max={1000} value={maxCost} placeholder={'До'} onChange={(e)=> dispatch(setMaxCost(+e.target.value))}/>
                            </div>
                        </div>
                        <div className={classes.inputBLock}>
                            <button data-testid={'open-more-options-btn'} type={'button'} onClick={()=>setMoreOptions(!moreOptions)} className={!moreOptions ? classes.openMoreFilter : classes.openMoreFilterActive}>
                                Больше опций
                                <SvgSettings width={'16'} height={'18'} color={'#664EF9'}/>
                            </button>
                        </div>
                        <Link className={classes.openMapButton} to={'/404'}>
                            На карте
                            <SvgLocation width={'15'} height={'15'} color={'#664EF9'}/>
                        </Link>
                        <button type={'button'} onClick={()=>catalogFilterHandler('flats')} className={classes.filterButton}>
                            Показать
                            <SvgArrowRight width={'9'} height={'16'} color={'#242424'}/>
                        </button>
                    </TabPanel>
                    <TabPanel data-testid={'tab-panel'}>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Город</p>
                            <MySelect options={cityOption} handler={(e)=>dispatch(setCity(e))} value={cityCurrentValue ? cityCurrentValue : 'Выберите'}/>
                        </div>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Комнаты</p>
                            <MySelect options={roomsOption} handler={(e)=>dispatch(setRooms(e))} value={roomsCurrentValue ? roomsCurrentValue : 'Выберите'}/>
                        </div>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Цена за сутки (BYN)</p>
                            <div className={classes.costBlock}>
                                <input type='number' min={0} max={1000} value={minCost} placeholder={'От'} onChange={(e)=> dispatch(setMinCost(+e.target.value))}/>
                                -
                                <input type='number' min={0} max={1000} value={maxCost} placeholder={'До'} onChange={(e)=> dispatch(setMaxCost(+e.target.value))}/>
                            </div>
                        </div>
                        <div className={classes.inputBLock}>
                            <button data-testid={'open-more-options-btn'} type={'button'} onClick={()=>setMoreOptions(!moreOptions)} className={!moreOptions ? classes.openMoreFilter : classes.openMoreFilterActive}>
                                Больше опций
                                <SvgSettings width={'16'} height={'18'} color={'#664EF9'}/>
                            </button>
                        </div>
                        <Link className={classes.openMapButton} to={'/404'}>
                            На карте
                            <SvgLocation width={'15'} height={'15'} color={'#664EF9'}/>
                        </Link>
                        <button type={'button'} onClick={()=>catalogFilterHandler('cottages')} className={classes.filterButton}>
                            Показать
                            <SvgArrowRight width={'9'} height={'16'} color={'#242424'}/>
                        </button>
                    </TabPanel>
                    <TabPanel data-testid={'tab-panel'}>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Город</p>
                            <MySelect options={cityOption} handler={(e)=>dispatch(setCity(e))} value={cityCurrentValue ? cityCurrentValue : 'Выберите'}/>
                        </div>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Комнаты</p>
                            <MySelect options={roomsOption} handler={(e)=>dispatch(setRooms(e))} value={roomsCurrentValue ? roomsCurrentValue : 'Выберите'}/>
                        </div>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Цена за сутки (BYN)</p>
                            <div className={classes.costBlock}>
                                <input type='number' min={0} max={1000} value={minCost} placeholder={'От'} onChange={(e)=> dispatch(setMinCost(+e.target.value))}/>
                                -
                                <input type='number' min={0} max={1000} value={maxCost} placeholder={'До'} onChange={(e)=> dispatch(setMaxCost(+e.target.value))}/>
                            </div>
                        </div>
                        <div className={classes.inputBLock}>
                            <button data-testid={'open-more-options-btn'} type={'button'} onClick={()=>setMoreOptions(!moreOptions)} className={!moreOptions ? classes.openMoreFilter : classes.openMoreFilterActive}>
                                Больше опций
                                <SvgSettings width={'16'} height={'18'} color={'#664EF9'}/>
                            </button>
                        </div>
                        <Link className={classes.openMapButton} to={'/404'}>
                            На карте
                            <SvgLocation width={'15'} height={'15'} color={'#664EF9'}/>
                        </Link>
                        <button type={'button'} onClick={()=>catalogFilterHandler('baths')} className={classes.filterButton}>
                            Показать
                            <SvgArrowRight width={'9'} height={'16'} color={'#242424'}/>
                        </button>
                    </TabPanel>
                    <TabPanel data-testid={'tab-panel'}>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Город</p>
                            <MySelect options={cityOption} handler={(e)=>dispatch(setCity(e))} value={cityCurrentValue ? cityCurrentValue : 'Выберите'}/>
                        </div>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Посадочные места</p>
                            <MySelect options={roomsOption} handler={(e)=>dispatch(setRooms(e))} value={roomsCurrentValue ? roomsCurrentValue : 'Выберите'}/>
                        </div>
                        <div className={classes.inputBLock}>
                            <p className={classes.inputBLockText}>Цена за сутки (BYN)</p>
                            <div className={classes.costBlock}>
                                <input type='number' min={0} max={1000} value={minCost} placeholder={'От'} onChange={(e)=> dispatch(setMinCost(+e.target.value))}/>
                                -
                                <input type='number' min={0} max={1000} value={maxCost} placeholder={'До'} onChange={(e)=> dispatch(setMaxCost(+e.target.value))}/>
                            </div>
                        </div>
                        <Link className={classes.openMapButton} to={'/404'}>
                            На карте
                            <SvgLocation width={'15'} height={'15'} color={'#664EF9'}/>
                        </Link>
                        <button type={'button'} onClick={()=>catalogFilterHandler('cars')} className={classes.filterButton}>
                            Показать
                            <SvgArrowRight width={'9'} height={'16'} color={'#242424'}/>
                        </button>
                    </TabPanel>
                </Tabs>
                {
                    moreOptions && tabIndex !== 3
                    && <MoreOptions />

                }

            </form>
        </section>
    );
};

export default MainFilterSection;
