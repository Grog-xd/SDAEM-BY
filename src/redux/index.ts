import {combineReducers} from 'redux';
import {configureStore} from '@reduxjs/toolkit';
import {useDispatch} from 'react-redux';

import mainPage from './mainPage';
import newsPage from './newsPage';
import loginPage from './loginPage';

const rootReducer = combineReducers({
    news: newsPage,
    main:mainPage,
    login: loginPage,
})

export const store = configureStore({
    reducer: rootReducer,
})


export type AppDispatch = typeof store.dispatch
export const useAppDispatch: () => AppDispatch = useDispatch
