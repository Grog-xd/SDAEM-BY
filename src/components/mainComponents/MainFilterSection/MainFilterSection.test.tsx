import {render, screen} from '@testing-library/react';
import {Provider} from 'react-redux';

import {MemoryRouter} from 'react-router-dom';

import userEvent from '@testing-library/user-event';

import {store} from '../../../redux';

import MainFilterSection from './MainFilterSection';

describe('main filter section tests', () =>{
    test('snapshot test', ()=>{
        render(
            <Provider store={store} >
                <MemoryRouter>
                    <MainFilterSection />
                </MemoryRouter>
            </Provider>,
        )
        const section = screen.getByTestId('main-filter-section')
        expect(section).toMatchSnapshot();
    })
    test('tabs test', () =>{
        render(
            <Provider store={store} >
                <MemoryRouter>
                    <MainFilterSection />
                </MemoryRouter>
            </Provider>,
        )
        const tabBtn = screen.getAllByTestId('tab-btn')
        const tabPanel = screen.getAllByTestId('tab-panel')

        expect(tabPanel[0]).toBeInTheDocument()
        userEvent.click(tabBtn[1])
        expect(tabPanel[1]).toBeInTheDocument()
        userEvent.click(tabBtn[2])
        expect(tabPanel[2]).toBeInTheDocument()
        userEvent.click(tabBtn[3])
        expect(tabPanel[3]).toBeInTheDocument()
    })

    test('more options test', () =>{
        render(
            <Provider store={store} >
                <MemoryRouter>
                    <MainFilterSection />
                </MemoryRouter>
            </Provider>,
        )

        const btn = screen.getAllByTestId('open-more-options-btn')

        expect(screen.queryByTestId('more-options')).not.toBeInTheDocument()

        userEvent.click(btn[0])

        expect(screen.getByTestId('more-options')).toBeInTheDocument()
    })
})
