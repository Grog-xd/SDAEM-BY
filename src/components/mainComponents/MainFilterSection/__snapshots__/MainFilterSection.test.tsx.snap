// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main filter section tests snapshot test 1`] = `
<section
  class="filterSection"
  data-testid="main-filter-section"
>
  <h1>
    Sdaem.by - у нас живут 
    <b>
      ваши объявления
    </b>
  </h1>
  <form
    class="mainForm"
  >
    <div
      class="react-tabs"
      data-rttabs="true"
    >
      <ul
        class="react-tabs__tab-list"
        role="tablist"
      >
        <li
          aria-controls="panel:r0:0"
          aria-disabled="false"
          aria-selected="true"
          class="react-tabs__tab react-tabs__tab--selected"
          data-rttab="true"
          data-testid="tab-btn"
          id="tab:r0:0"
          role="tab"
          tabindex="0"
        >
          Квартиры на сутки
        </li>
        <li
          aria-controls="panel:r0:1"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          data-testid="tab-btn"
          id="tab:r0:1"
          role="tab"
        >
          Коттеджи и усадьбы
        </li>
        <li
          aria-controls="panel:r0:2"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          data-testid="tab-btn"
          id="tab:r0:2"
          role="tab"
        >
          Бани и сауны
        </li>
        <li
          aria-controls="panel:r0:3"
          aria-disabled="false"
          aria-selected="false"
          class="react-tabs__tab"
          data-rttab="true"
          data-testid="tab-btn"
          id="tab:r0:3"
          role="tab"
        >
          Авто напрокат
        </li>
      </ul>
      <div
        aria-labelledby="tab:r0:0"
        class="react-tabs__tab-panel react-tabs__tab-panel--selected"
        data-testid="tab-panel"
        id="panel:r0:0"
        role="tabpanel"
      >
        <div
          class="inputBLock"
        >
          <p
            class="inputBLockText"
          >
            Город
          </p>
          <div
            class="select "
          >
            <button
              class="changeBlock"
              type="button"
            >
              <div>
                <p
                  class="textValue"
                >
                  Выберите
                </p>
              </div>
              <svg
                fill="none"
                height="16"
                viewBox="0 0 12 7"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.5 1.38477L6 5.64152L10.5 1.38477"
                  stroke="#4E64F9"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="inputBLock"
        >
          <p
            class="inputBLockText"
          >
            Комнаты
          </p>
          <div
            class="select "
          >
            <button
              class="changeBlock"
              type="button"
            >
              <div>
                <p
                  class="textValue"
                >
                  Выберите
                </p>
              </div>
              <svg
                fill="none"
                height="16"
                viewBox="0 0 12 7"
                width="14"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M1.5 1.38477L6 5.64152L10.5 1.38477"
                  stroke="#4E64F9"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
        <div
          class="inputBLock"
        >
          <p
            class="inputBLockText"
          >
            Цена за сутки (BYN)
          </p>
          <div
            class="costBlock"
          >
            <input
              max="1000"
              min="0"
              placeholder="От"
              type="number"
              value="0"
            />
            -
            <input
              max="1000"
              min="0"
              placeholder="До"
              type="number"
              value="1000"
            />
          </div>
        </div>
        <div
          class="inputBLock"
        >
          <button
            class="openMoreFilter"
            data-testid="open-more-options-btn"
            type="button"
          >
            Больше опций
            <svg
              fill="none"
              height="18"
              viewBox="0 0 16 18"
              width="16"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.91109 4.29814C5.91109 2.90551 4.937 1.73697 3.63453 1.4358V0.661897C3.63453 0.296309 3.33822 0 2.97264 0C2.60705 0 2.31074 0.296309 2.31074 0.661897V1.4358C1.00835 1.73689 0.0341797 2.90551 0.0341797 4.29814C0.0341797 5.69077 1.00827 6.85931 2.31074 7.16048V17.3381C2.31074 17.7037 2.60705 18 2.97264 18C3.33822 18 3.63453 17.7037 3.63453 17.3381V7.16048C4.937 6.85931 5.91109 5.69077 5.91109 4.29814ZM1.35805 4.29814C1.35805 3.40781 2.08238 2.68348 2.97271 2.68348C3.86303 2.68348 4.58737 3.40781 4.58737 4.29814C4.58737 5.18846 3.86303 5.9128 2.97271 5.9128C2.08238 5.9128 1.35805 5.18846 1.35805 4.29814Z"
                fill="#664EF9"
              />
              <path
                d="M8.66188 8.89222V0.661897C8.66188 0.296309 8.36557 0 7.99998 0C7.63439 0 7.33808 0.296309 7.33808 0.661897V8.89222C6.03569 9.19331 5.06152 10.3619 5.06152 11.7546C5.06152 13.1472 6.03562 14.3157 7.33808 14.6169V17.3381C7.33808 17.7037 7.63439 18 7.99998 18C8.36557 18 8.66188 17.7037 8.66188 17.3381V14.6169C9.96427 14.3158 10.9384 13.1472 10.9384 11.7546C10.9384 10.3619 9.96434 9.19338 8.66188 8.89222ZM6.38539 11.7546C6.38539 10.8642 7.10973 10.1399 8.00005 10.1399C8.89038 10.1399 9.61471 10.8642 9.61471 11.7546C9.61471 12.6449 8.89038 13.3692 8.00005 13.3692C7.10973 13.3692 6.38539 12.6449 6.38539 11.7546Z"
                fill="#664EF9"
              />
              <path
                d="M15.9658 7.20151C15.9658 5.80888 14.9917 4.64034 13.6892 4.33918V0.661897C13.6892 0.296309 13.3929 0 13.0273 0C12.6617 0 12.3654 0.296309 12.3654 0.661897V4.33918C11.063 4.64027 10.0889 5.80888 10.0889 7.20151C10.0889 8.59415 11.063 9.76269 12.3654 10.0639V17.3381C12.3654 17.7037 12.6617 18 13.0273 18C13.3929 18 13.6892 17.7037 13.6892 17.3381V10.0639C14.9917 9.76269 15.9658 8.59415 15.9658 7.20151ZM11.4127 7.20151C11.4127 6.31119 12.1371 5.58685 13.0274 5.58685C13.9177 5.58685 14.6421 6.31119 14.6421 7.20151C14.6421 8.09184 13.9177 8.81618 13.0274 8.81618C12.1371 8.81618 11.4127 8.09184 11.4127 7.20151Z"
                fill="#664EF9"
              />
            </svg>
          </button>
        </div>
        <a
          class="openMapButton"
          href="/404"
        >
          На карте
          <svg
            fill="none"
            height="15"
            viewBox="0 0 9 10"
            width="15"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.43806 1.74419C6.7055 0.627907 5.4962 0 4.13573 0C2.7869 0 1.5776 0.627907 0.821781 1.74419C0.0659671 2.83721 -0.108452 4.23256 0.356665 5.45349C0.484572 5.77907 0.682246 6.11628 0.93806 6.4186L3.87992 9.88372C3.94969 9.95349 4.01946 10 4.12411 10C4.22876 10 4.29853 9.95349 4.36829 9.88372L7.32178 6.4186C7.5776 6.11628 7.7869 5.7907 7.90318 5.45349C8.36829 4.23256 8.19387 2.83721 7.43806 1.74419ZM4.13573 5.86047C3.13573 5.86047 2.31015 5.03488 2.31015 4.03488C2.31015 3.03488 3.13573 2.2093 4.13573 2.2093C5.13574 2.2093 5.96132 3.03488 5.96132 4.03488C5.96132 5.03488 5.14736 5.86047 4.13573 5.86047Z"
              fill="#664EF9"
            />
          </svg>
        </a>
        <button
          class="filterButton"
          type="button"
        >
          Показать
          <svg
            fill="none"
            height="16"
            viewBox="0 0 9 16"
            width="9"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M1 14.8574L7.57143 8.28599L0.999999 1.71456"
              stroke="#242424"
            />
          </svg>
        </button>
      </div>
      <div
        aria-labelledby="tab:r0:1"
        class="react-tabs__tab-panel"
        data-testid="tab-panel"
        id="panel:r0:1"
        role="tabpanel"
      />
      <div
        aria-labelledby="tab:r0:2"
        class="react-tabs__tab-panel"
        data-testid="tab-panel"
        id="panel:r0:2"
        role="tabpanel"
      />
      <div
        aria-labelledby="tab:r0:3"
        class="react-tabs__tab-panel"
        data-testid="tab-panel"
        id="panel:r0:3"
        role="tabpanel"
      />
    </div>
  </form>
</section>
`;
